Cauli/Florets/Inspector/Record List/InspectorTableViewController.swift

```diff
@@ -78,26 +83,75 @@ internal class InspectorTableViewController: UITableViewController {
         let distanceToBottom = scrollView.contentSize.height - scrollView.frame.height - scrollView.contentOffset.y
         guard !scrolledToEnd, !isLoading, distanceToBottom < 100 else { return }
         isLoading = true
-        let newRecords = cauli.storage.records(InspectorTableViewController.recordPageSize, after: records.last)
+        let newRecords = cauli.storage.records(InspectorTableViewController.recordPageSize, after: dataSource.items.last)
         if newRecords.isEmpty {
             isLoading = false
             scrolledToEnd = true
         } else if #available(iOS 11, *) {
             tableView.performBatchUpdates({
-                let indexPaths = (records.count..<(records.count + newRecords.count)).map { IndexPath(row: $0, section: 0) }
+                let numberOfExistingRecords = dataSource.filteredItems.count
+                let numberOfAddedRecords = dataSource.filter(records: newRecords).count
+                let indexPaths = (numberOfExistingRecords..<(numberOfExistingRecords + numberOfAddedRecords)).map { IndexPath(row: $0, section: 0) }
                 tableView.insertRows(at: indexPaths, with: .bottom)
-                records.append(contentsOf: newRecords)
+                dataSource.append(records: newRecords)
             }, completion: { [weak self] _ in
                 self?.isLoading = false
             })
         } else {
-            let indexPaths = (records.count..<(records.count + newRecords.count)).map { IndexPath(row: $0, section: 0) }
+            let numberOfExistingRecords = dataSource.filteredItems.count
+            let numberOfAddedRecords = dataSource.filter(records: newRecords).count
+            let indexPaths = (numberOfExistingRecords..<(numberOfExistingRecords + numberOfAddedRecords)).map { IndexPath(row: $0, section: 0) }
             tableView.beginUpdates()
             tableView.insertRows(at: indexPaths, with: .bottom)
-            records.append(contentsOf: newRecords)
+            dataSource.append(records: newRecords)
             tableView.endUpdates()
             self.isLoading = false
         }
     }
 
 }
+
+// MARK: - UISearchResultsUpdating
+
+extension InspectorTableViewController: UISearchResultsUpdating {
+
+    func updateSearchResults(for searchController: UISearchController) {
+        let searchString = searchController.searchBar.text ?? ""
+        let searchTerms = searchString.trimmingCharacters(in: CharacterSet.whitespaces).components(separatedBy: " ")
```

Comment:
We didn't talk about this before, but I am still interested: What is your usecase here to have multiple terms? I am not super sure this is going to be the default way. Your implementation of multiple terms here is searching if any of the words are matching. Isn't it rather both? Is it possible to define a "default" behavior here or should we rather (for now) remove the searching for multiple terms and interpret the whole string as one search term?

Reply:
The use case was to be able to search for more than one term at the same time. I think you are right that the more common case would be the AND operator instead of OR.

Would you prefer me to rewrite it to AND or to remove the feature and only use the search string? For the first step we may not need to do a complex solution until a real usecase arises. 
One I could imagine would be "filter for all requests from example.com that have a status code of 404", but for that we would need to extend filtering on other fields as well.

Reply:
Yep, removed. We can always extend it if an actual use case comes up.